src/snowflake/snowpark/session.py

```diff
@@ -717,11 +734,23 @@ def _resolve_packages(
                         f"and add it via session.add_import()."
                     )
                 elif not use_local_version:
+                    # check if user as requested for a specific version of package 'pkg==version'
```

Comment:
Why is this check only under `use_local_version`?

Reply:
Updated to cover all cases